### DISCUSSION TITLE
Operation of start_offset and end_offset

### DISCUSSION THREAD

**Discussion by raiAmagi** [dagster-io/dagster#25216] [Q&A]
I would like to know about “Overriding default dependency rules” in “Partitioning assets”.
(https://docs.dagster.io/concepts/partitions-schedules-sensors/partitioning-assets#default-partition-dependency-rules)

In the sample PG, I have changed the mapping like TimeWindowPartitionMapping(start_offset=-1, end_offset=-1)
The mapping is changed.

In this case, the “2024-10-06” partition in the downstream asset has a dependency on the “2024-10-05” and “2024-10-06” partitions in the upstream asset.
The dependency relationship occurs between the two partitions.

The dependency relationship will be as follows.
Upstream      “2024-10-05”　　　“2024-10-06
　　　　　　　　　| 　　　　　　　|
Downstream “2024-10-06” ---------

For example
　The upstream asset “2024-10-05” is not materialized.
　The upstream asset “2024-10-06” has been materialized.
Materialize downstream asset “2024-10-06” in this state.

In this case, we thought that a warning would be generated because the upstream asset “2024-10-05” has not been materialized.

However, materialization can be performed without warning or error.

I materialize manually with the UI.

Is my understanding wrong?

Also, I would like to know more about start_offset and end_offset in the API document, including how they work.

use version : 1.8.11

I am using a translation site to write the sentences.
Sorry if the text is rude.

**Comment by smackesey** [comment]
Hi @raiAmagi, would you mind providing a simple self-contained code sample with two assets and your defined `PartitionMapping`? That will help answer this question without any misunderstanding.

**Comment by raiAmagi** [comment]
Thanks for the reply.

Here is the PG that we tried.

------------------------------------------------------------------------

```
from dagster import (
    AssetDep,
    DailyPartitionsDefinition,
    TimeWindowPartitionMapping,
    asset,
)

partitions_def = DailyPartitionsDefinition(start_date="2023-01-21")


@asset(partitions_def=partitions_def, group_name='partition_deps_rules_override', )
def partition_rules_override_events(): ...


@asset(
    partitions_def=partitions_def,
    deps=[
        AssetDep(
            partition_rules_override_events,
            partition_mapping=TimeWindowPartitionMapping(
                start_offset=-1, end_offset=-1
            ),
        )
    ],
    group_name='partition_deps_rules_override'
)
def partition_rules_override_yesterday_event_stats(): ...
```
